client/index.js

```diff
@@ -50,11 +49,31 @@ envConfig.setConfig({
 const asPath = getURL()
 
 const pageLoader = new PageLoader(buildId, prefix)
-window.__NEXT_LOADED_PAGES__.forEach(([r, f]) => {
-  pageLoader.registerPage(r, f)
+if (page === '_error') {
+  pageLoader.registerPage(htmlescape(pathname), () => {
+    const e = new Error(`Page does not exist: ${htmlescape(pathname)}`)
+    e.statusCode = 404
+    return { error: e }
+  })
+}
+
+window._routes = new Proxy(window._routes, {
```

Comment:
There's no polyfill for `Proxy` and this won't work on IE11.

Reply:
https://github.com/zeit/next.js/pull/4943#issuecomment-424948950